packages/cloud_firestore/cloud_firestore_web/lib/src/utils/decode_utility.dart

```diff
@@ -35,12 +35,16 @@ class DecodeUtility {
   /// Decodes an incoming value to its proper type.
   static dynamic valueDecode(
       dynamic value, FirebaseFirestorePlatform firestore) {
+    // Cannot be done with Dart 3.2 constraints
+    // ignore: invalid_runtime_check_with_js_interop_types
     if (value is JSObject &&
         value.instanceof(GeoPointConstructor as JSFunction)) {
       return GeoPoint((value as GeoPointJsImpl).latitude.toDartDouble,
           (value as GeoPointJsImpl).longitude.toDartDouble);
     } else if (value is DateTime) {
       return Timestamp.fromDate(value);
+      // Cannot be done with Dart 3.2 constraints
+      // ignore: invalid_runtime_check_with_js_interop_types
```

Comment:
But this is invalid code, I think! Runtime behavior will be wrong w/ wasm!

Reply:
Drive-by comment:

It looks like both Dart values and JS values flow to `value`. In this case, I'd recommend checking all the Dart values first (`is DateTime`, etc.). After that, I'd cast `value` to a `JSAny`, and then do `isA` checks with the resulting `JSAny`. As an example:

```dart
if (value is DateTime) {
  ...
} else {
  final jsValue = value as JSAny?;
  if (jsValue.isA<GeoPointJsImpl>()) {
    ...
  } else if (jsValue.isA<JSArray>()) {
    List<JSAny?> list = (jsValue as JSArray).toDart;
    ...
  }
}
```

It also looks like there's a check against `List<dynamic>`. If `value` here is actually a JS array, use `isA<JSArray>`, as `is List<dynamic>` may be true when compiling to JS, but false when compiling to Wasm. The general guidance is that if a value is a JS value, use interop to type-check it and not Dart `is` checks.